presto-hive/src/test/java/io/prestosql/plugin/hive/HiveQueryRunner.java

```diff
@@ -269,6 +281,60 @@ private static void copyTableBucketed(QueryRunner queryRunner, QualifiedObjectNa
         log.info("Imported %s rows for %s in %s", rows, table.getObjectName(), nanosSince(start).convertToMostSuccinctTimeUnit());
     }
 
+    public static Session createPartitionedSession(Optional<SelectedRole> role)
+    {
+        return testSessionBuilder()
+                .setIdentity(Identity.forUser("hive")
+                        .withRoles(role.map(selectedRole -> ImmutableMap.of("hive", selectedRole))
+                                .orElse(ImmutableMap.of()))
+                        .build())
+                .setCatalog(HIVE_PARTITIONED_CATALOG)
+                .setSchema(TPCH_PARTITIONED_SCHEMA)
+                .build();
+    }
+
+    public static void copyTpchTablesPartitioned(
+            QueryRunner queryRunner,
+            String sourceCatalog,
+            String sourceSchema,
+            Session session,
+            Iterable<TpchTable<?>> tables)
+    {
+        log.info("Loading data from %s.%s...", sourceCatalog, sourceSchema);
+        long startTime = System.nanoTime();
+        for (TpchTable<?> table : tables) {
+            copyTablePartitioned(queryRunner, new QualifiedObjectName(sourceCatalog, sourceSchema, table.getTableName().toLowerCase(ENGLISH)), session);
+        }
+        log.info("Loading from %s.%s complete in %s", sourceCatalog, sourceSchema, nanosSince(startTime).toString(SECONDS));
+    }
+
+    private static void copyTablePartitioned(QueryRunner queryRunner, QualifiedObjectName table, Session session)
+    {
+        long start = System.nanoTime();
+        log.info("Running import for %s", table.getObjectName());
+        @Language("SQL") String sql;
+        switch (table.getObjectName()) {
+            case "orders":
+            case "part":
+            case "partsupp":
+            case "supplier":
+            case "nation":
+            case "region":
+                sql = format("CREATE TABLE %s AS SELECT * FROM %s", table.getObjectName(), table);
+                break;
+            case "lineitem":
+                sql = format("CREATE TABLE %s WITH (partitioned_by=array['suppkey']) AS SELECT orderkey, partkey, linenumber, quantity, extendedprice, discount, tax, returnflag, linestatus, shipdate, commitdate, receiptdate, shipinstruct, shipmode, comment, suppkey FROM %s", table.getObjectName(), table);
```

Comment:
please partition by `shiptdate` which is more natural

Reply:
Partitioning on shiptdate gives ~2.5K partitions which makes the test setup very slow (triggers major GC a lot on my laptop). Partitioning suppkey gives 100 partitions which works well for testing.

Reply:
Ok. Since this is for just single test, Let's just create appropriate tables in test itself instead of using artificial keys

Reply:
Done